awx/ui_next/src/screens/Template/WorkflowJobTemplateVisualizer/Modals/NodeModals/NodeModal.jsx

```diff
@@ -0,0 +1,213 @@
+} from '@patternfly/react-core';
+import Wizard from '@components/Wizard';
+import { NodeTypeStep } from './NodeTypeStep';
+import { RunStep, NodeNextButton } from '.';
+
+function NodeModal({ askLinkType, history, i18n, onSave, title }) {
+  const dispatch = useContext(WorkflowDispatchContext);
+  const { nodeToEdit } = useContext(WorkflowStateContext);
+
+  let defaultApprovalDescription = '';
+  let defaultApprovalName = '';
+  let defaultApprovalTimeout = 0;
+  let defaultNodeResource = null;
+  let defaultNodeType = 'job_template';
+  if (nodeToEdit && nodeToEdit.unifiedJobTemplate) {
+    if (
+      nodeToEdit &&
+      nodeToEdit.unifiedJobTemplate &&
+      (nodeToEdit.unifiedJobTemplate.type ||
+        nodeToEdit.unifiedJobTemplate.unified_job_type)
+    ) {
+      const ujtType =
+        nodeToEdit.unifiedJobTemplate.type ||
+        nodeToEdit.unifiedJobTemplate.unified_job_type;
+      switch (ujtType) {
+        case 'job_template':
+        case 'job':
+          defaultNodeType = 'job_template';
+          defaultNodeResource = nodeToEdit.unifiedJobTemplate;
+          break;
+        case 'project':
+        case 'project_update':
+          defaultNodeType = 'project_sync';
+          defaultNodeResource = nodeToEdit.unifiedJobTemplate;
+          break;
+        case 'inventory_source':
+        case 'inventory_update':
+          defaultNodeType = 'inventory_source_sync';
+          defaultNodeResource = nodeToEdit.unifiedJobTemplate;
+          break;
+        case 'workflow_job_template':
+        case 'workflow_job':
+          defaultNodeType = 'workflow_job_template';
+          defaultNodeResource = nodeToEdit.unifiedJobTemplate;
+          break;
+        case 'workflow_approval_template':
+        case 'workflow_approval':
+          defaultNodeType = 'approval';
+          defaultApprovalName = nodeToEdit.unifiedJobTemplate.name;
+          defaultApprovalDescription =
+            nodeToEdit.unifiedJobTemplate.description;
+          defaultApprovalTimeout = nodeToEdit.unifiedJobTemplate.timeout;
+          break;
+        default:
+      }
+    }
+  }
+  const [approvalDescription, setApprovalDescription] = useState(
+    defaultApprovalDescription
+  );
+  const [approvalName, setApprovalName] = useState(defaultApprovalName);
+  const [approvalTimeout, setApprovalTimeout] = useState(
+    defaultApprovalTimeout
+  );
+  const [linkType, setLinkType] = useState('success');
+  const [nodeResource, setNodeResource] = useState(defaultNodeResource);
+  const [nodeType, setNodeType] = useState(defaultNodeType);
+  const [triggerNext, setTriggerNext] = useState(0);
+
+  const clearQueryParams = () => {
+    const parts = history.location.search.replace(/^\?/, '').split('&');
+    const otherParts = parts.filter(param =>
+      /^!(job_templates\.|projects\.|inventory_sources\.|workflow_job_templates\.)/.test(
+        param
+      )
+    );
+    history.push(`${history.location.pathname}?${otherParts.join('&')}`);
```

Comment:
Whenever I add a new node, this error shows up in the console: 
<img width="676" alt="Screen Shot 2020-01-28 at 5 05 12 PM" src="https://user-images.githubusercontent.com/15881645/73311094-afc4f380-41f3-11ea-9dee-1ca649a71757.png">

You could swap history.push for history.replace to get around this - it replaces the current entry on the history stack.